File: platform/platform-api/src/main/java/com/yazino/platform/tournament/TournamentVariationRound.java
package com.yazino.platform.tournament;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.apache.commons.lang3.builder.HashCodeBuilder;
import org.apache.commons.lang3.builder.ToStringBuilder;

import java.io.Serializable;
import java.math.BigDecimal;

import static org.apache.commons.lang3.Validate.notNull;

public class TournamentVariationRound implements Serializable {
    private static final long serialVersionUID = -3672223520978262638L;
    private final int roundNumber;
    private final long roundEndInterval;
    private final long roundLength;
    private final BigDecimal minimumBalance;
    private final BigDecimal gameVariationTemplateId;
    private final String clientPropertiesId;
    private final String description;

    public TournamentVariationRound(final int roundNumber,
                                    final long roundEndInterval,
                                    final long roundLength,
                                    final BigDecimal gameVariationTemplateId,
                                    final String clientPropertiesId,
                                    final BigDecimal minimumBalance,
                                    final String description) {
        notNull(gameVariationTemplateId, "Game Variation Template Id may not be Null.");

        this.roundNumber = roundNumber;
        this.roundEndInterval = roundEndInterval;
        this.roundLength = roundLength;
        this.gameVariationTemplateId = gameVariationTemplateId;
        this.clientPropertiesId = clientPropertiesId;
        this.minimumBalance = minimumBalance;
        this.description = description;
    }

    public int getRoundNumber() {
        return roundNumber;
    }

    public long getRoundEndInterval() {
        return roundEndInterval;
    }

    public long getRoundLength() {
        return roundLength;
    }

    public BigDecimal getGameVariationTemplateId() {
        return gameVariationTemplateId;
    }

    public String getClientPropertiesId() {
        return clientPropertiesId;
    }

    public BigDecimal getMinimumBalance() {
        return minimumBalance;
    }

    public String getDescription() {
        return description;
    }

    @Override
    public boolean equals(final Object obj) {
        if (obj == null) {
            return false;
        }
        if (obj == this) {
            return true;
        }
        if (obj.getClass() != getClass()) {
            return false;
        }
        final TournamentVariationRound rhs = (TournamentVariationRound) obj;
        return new EqualsBuilder()
                .append(roundEndInterval, rhs.roundEndInterval)
                .append(roundLength, rhs.roundLength)
                .append(roundNumber, rhs.roundNumber)
                .append(clientPropertiesId, rhs.clientPropertiesId)
                .append(gameVariationTemplateId, rhs.gameVariationTemplateId)
                .append(minimumBalance, rhs.minimumBalance)
                .append(description, rhs.description)
                .isEquals();
    }

    @Override
    public int hashCode() {
        return new HashCodeBuilder(13, 17)
                .append(roundEndInterval)
                .append(roundLength)
                .append(roundNumber)
                .append(clientPropertiesId)
                .append(gameVariationTemplateId)
                .append(minimumBalance)
                .append(description)
                .toHashCode();
    }

    @Override
    public String toString() {
        return new ToStringBuilder(this)
                .append(roundEndInterval)
                .append(roundLength)
                .append(roundNumber)
                .append(clientPropertiesId)
                .append(gameVariationTemplateId)
                .append(minimumBalance)
                .append(description)
                .toString();
    }
}

File: platform/control-centre/src/main/scala/strata/server/lobby/controlcentre/model/TournamentVariationRound.scala
package strata.server.lobby.controlcentre.model

import org.apache.commons.lang3.builder.{HashCodeBuilder, EqualsBuilder, ToStringBuilder}
import com.yazino.platform.tournament.{TournamentVariationRound => PlatformRound}
import java.math

class TournamentVariationRound(val id: BigDecimal,
                               val number: Int,
                               val endInterval: Int,
                               val length: Int,
                               val gameVariationId: BigDecimal,
                               val clientPropertiesId: String,
                               val minimumBalance: BigDecimal,
                               val description: String) {

    def withId(newId: BigDecimal): TournamentVariationRound =
        new TournamentVariationRound(newId,
            number,
            endInterval,
            length,
            gameVariationId,
            clientPropertiesId,
            minimumBalance,
            description)

    def toPlatform: PlatformRound = new PlatformRound(number, endInterval, length, toJava(gameVariationId),
        clientPropertiesId, toJava(minimumBalance), description)

    private def toJava(number: BigDecimal): math.BigDecimal = if (number != null) number.underlying() else null

    override def toString: String = new ToStringBuilder(this)
        .append(id)
        .append(number)
        .append(endInterval)
        .append(length)
        .append(gameVariationId)
        .append(clientPropertiesId)
        .append(minimumBalance)
        .append(description)
        .toString

    override def equals(obj: Any): Boolean = obj match {
        case other: TournamentVariationRound => other.getClass == getClass &&
            new EqualsBuilder()
                .append(id, other.id)
                .append(number, other.number)
                .append(endInterval, other.endInterval)
                .append(length, other.length)
                .append(gameVariationId, other.gameVariationId)
                .append(clientPropertiesId, other.clientPropertiesId)
                .append(minimumBalance, other.minimumBalance)
                .append(description, other.description)
                .isEquals
        case _ => false
    }

    override def hashCode: Int = new HashCodeBuilder()
        .append(id)
        .append(number)
        .append(endInterval)
        .append(length)
        .append(gameVariationId)
        .append(clientPropertiesId)
        .append(minimumBalance)
        .append(description)
        .hashCode

}
